src/service.py

```diff
@@ -176,7 +185,36 @@ def cluster_sentences(self, language_name: str, save_path: str, sentences: List[
         if no_n_input:
             examples = recommend_sentences
 
-        return examples, recommend_sentences
+        return examples, recommend_sentences, sentences, best_labels
+
+    def kwic(self, selword: str, sentence_with_pos: list):
+        """
+        :param: selword
+        :param: sentenceWithPOS
+
+        sentence_with_pos examples:
+        [("NOUN", "bank", ["I go to the bank", "The house lies the right of the river bank"]),
+        ("VERB", "bank", ["I banked in a slot"])
+        """
+        # This is similar to sentenceWithPOS but processed after KWIC
+        result = []
+        for sentTuple in sentence_with_pos:
+            sents_kwic = []
+            result.append((sentTuple[0], sentTuple[1], sentTuple[2], sents_kwic))
+
+            sents_origin = sentTuple[2]
+            for sent in sents_origin:
+                # result_text = keywords_in_context(sent, [selword])
+                # Highlight Keywords
+                # result_text = find_and_replace(result_text, selword, "\x1b[34m" + selword + "\x1b[0m")
+                # sents_kwic.append(result_text)
+                window_words = get_keyword_window(selword, sent.split(" "))
```

Comment:
we can merge these two functions in a single function which I uploaded in my last pull request for more efficiency.